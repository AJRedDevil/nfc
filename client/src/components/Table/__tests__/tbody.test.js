// npm packages
import React from 'react';
import renderer from 'react-test-renderer';
import {range} from 'lodash';

// our packages
import TBody, {TableHeader, TableData, TableRow} from '../tbody';

describe('# TBody', () => {
  test('TableHeader', () => {
    const tree = renderer.create(<TableHeader data="SN" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('TableData', () => {
    const tree = renderer.create(<TableData data="1" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('TableRow', () => {
    const data = [1, 'test1', 'test2', 'test3'];
    const tree = renderer.create(<TableRow data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('TBody', () => {
    const headers = ['SN', 'Message'];
    const content = range(1, 5).map(i => [i, `test${i}`]);
    const data = [headers, ...content];
    const tree = renderer.create(<TBody data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
