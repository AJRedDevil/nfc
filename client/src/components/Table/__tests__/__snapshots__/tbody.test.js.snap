// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# TBody TBody 1`] = `
<tbody>
  <tr>
    <th>
      SN
    </th>
    <td>
      Message
    </td>
  </tr>
  <tr>
    <th>
      1
    </th>
    <td>
      test1
    </td>
  </tr>
  <tr>
    <th>
      2
    </th>
    <td>
      test2
    </td>
  </tr>
  <tr>
    <th>
      3
    </th>
    <td>
      test3
    </td>
  </tr>
  <tr>
    <th>
      4
    </th>
    <td>
      test4
    </td>
  </tr>
</tbody>
`;

exports[`# TBody TableData 1`] = `
<td>
  1
</td>
`;

exports[`# TBody TableHeader 1`] = `
<th>
  SN
</th>
`;

exports[`# TBody TableRow 1`] = `
<tr>
  <th>
    1
  </th>
  <td>
    test1
  </td>
  <td>
    test2
  </td>
  <td>
    test3
  </td>
</tr>
`;
